Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -63,28 +63,34 @@ Note: if the application is equipped with anti-reverse engineering controls, the
+
+When debug symbols are placed in a separate dSYM file, these symbols do not make to the final binary delivered to an end user. While the debug symbols embedded in the compiled binary, they can end up being present in the final binary and thus should be verified before the release.
 
 ### Static Analysis
 
-Use gobjdump to inspect the main binary and any included dylibs for Stabs and DWARF symbols.
+Use `objdump --syms` to inspect the main binary and other included dylibs, and look for symbols with 'd' (debug) flag associated with them. Check [objdump man page](https://www.unix.com/man-page/osx/1/objdump/ "objdump man page") for information about various other symbol's flag characters. A typical output for a binary containing debug symbols looks like following:
 
 ```bash
-$ gobjdump --stabs --dwarf TargetApp
-In archive MyTargetApp:
-
-armv5te:     file format mach-o-arm
+$ objdump --syms TargetApp
 
-aarch64:     file format mach-o-arm64
+0000000100007dc8 l    d  *UND* -[ViewController handleSubmitButton:]
+000000010000809c l    d  *UND* -[ViewController touchesBegan:withEvent:]
+0000000100008158 l    d  *UND* -[ViewController viewDidLoad]
+...
+000000010000916c l    d  *UND* _disable_gdb
+00000001000091d8 l    d  *UND* _detect_injected_dylds
+00000001000092a4 l    d  *UND* _isDebugged
+...
 ```
 
-Gobjdump is part of [binutils](https://www.gnu.org/s/binutils/ "Binutils") and can be installed on macOS via Homebrew.
+It is important to note, a binary without debug symbols will not have symbols with 'd' flag associated.
 
-Make sure that debugging symbols are stripped when the application is being built for production. Stripping debugging symbols will reduce the size of the binary and increase the difficulty of reverse engineering. To strip debugging symbols, set `Strip Debug Symbols During Copy` to `YES` via the project's build settings.
+`Objdump` from [binutils](https://www.gnu.org/s/binutils/ "Binutils") or [llvm-objdump](https://llvm.org/docs/CommandGuide/llvm-objdump.html "llvm-objdump") can be used. The output from the two may have subtle differences, but output from both will contain flag characters for symbols.
 
-A proper [Crash Reporter System](https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/AnalyzingCrashReports/AnalyzingCrashReports.html "Crash Reporter System") is possible because the system doesn't require any symbols in the application binary.
+To remove the embedded debugging symbols, ensure debug symbols stripping is enabled in Xcode configuration. To strip debugging symbols, set `Strip Debug Symbols During Copy` to `YES` via the project's build settings. Stripping debugging symbols will not only reduce the size of the binary and but also increase the difficulty of reverse engineering.
```

Comment:
```suggestion
To prevent the inclusion of debug symbols, set `Strip Debug Symbols During Copy` to `YES` via the XCode project's build settings. Stripping debugging symbols will not only reduce the size of the binary and but also increase the difficulty of reverse engineering.
```